apps/console/app/routes/_shell.agent.$agentSlug.branch.$branchSlug.config/route.tsx

```
@@ -0,0 +1,206 @@
+
+import { useState } from "react";
+import { useRouteLoaderData, useParams } from "react-router";
+import { object, string, nonEmpty, pipe, trim, message } from "valibot";
+import { parseWithValibot } from "@conform-to/valibot";
+
+import supportedModels from "@hebo/shared-data/json/supported-models";
+
+import { Button } from "@hebo/shared-ui/components/Button";
+import { Collapsible, CollapsibleContent, CollapsibleTrigger } from "@hebo/shared-ui/components/Collapsible";
+import { Card, CardContent } from "@hebo/shared-ui/components/Card";
+import { Badge } from "@hebo/shared-ui/components/Badge";
+import { BranchModelForm } from "./form";
+import { api } from "~console/lib/data";
+import { RailSymbol } from "lucide-react";
+
+import type { Route } from "./+types/route";
+
+const getModelDisplayName = (modelName: string): string => {
+  if (modelName === "custom") {
+    return "Custom Model";
+  }
+  
+  const model = supportedModels.find((m) => m.name === modelName);
+  return model?.displayName || modelName;
+};
+
+const BranchConfigSchema = object({
+  alias: message(pipe(string(), trim(), nonEmpty()), "Please enter a model alias"),
+  modelType: message(pipe(string(), trim(), nonEmpty()), "Please select a model type"),
+});
+
+export async function clientAction({ request, params }: Route.ClientActionArgs) {
+  const formData = await request.formData();
+  const action = String(formData.get("intent") || formData.get("_action") || "");
+
+  let alias = "";
+  let modelType = "";
+
+  if (action === "remove") {
+    alias = String(formData.get("alias") ?? "");
+    if (!alias.trim()) {
+      return { error: "Alias is required to remove a model" };
+    }
+  } else {
+    const submission = parseWithValibot(formData, { schema: BranchConfigSchema });
+    if (submission.status !== "success") {
+      return { lastResult: submission.reply() };
+    }
+    alias = String(submission.payload.alias ?? "");
+    modelType = String(submission.payload.modelType ?? "");
```

Stop providing empty fallback values, which will hide programming errors. You already validated the form result.

---

Addressed.